WebApiCoreSeed.WebApi/Controllers/UserController.cs

```diff
@@ -0,0 +1,110 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Threading.Tasks;
+using WebApiCoreSeed.Data.Models;
+using WebApiCoreSeed.Domain.Services.Interfaces;
+using WebApiCoreSeed.WebApi.Controllers.Dtos;
+
+namespace WebApiCoreSeed.WebApi.Controllers
+{
+    [Route("api/users")]
+    public class UserController : Controller
+    {
+        private readonly IUserService _userService;
+
+        public UserController(IUserService userService)
+        {
+            _userService = userService;
+        }
+
+        // GET api/users
+        [HttpGet]
+        public async Task<IActionResult> GetAll()
+        {
+            return Ok(await _userService.GetAsync());
+        }
+
+        // GET api/users/0E95A953-D023-4425-ED35-08D4E34D4DB8
+        [HttpGet("{id}")]
+        public async Task<IActionResult> Get(Guid id)
+        {
+            var user = await _userService.GetByIdAsync(id);
+            if (user == null)
+            {
+                return NotFound();
+            }
+
+            return Ok(user);
+        }
+
+        // POST api/users
+        [HttpPost]
+        public async Task<IActionResult> Create([FromBody]UserDto user)
+        {
+            if (user == null)
+            {
+                return BadRequest();
+            }
+
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }

```

Comment:
for these validations, what do you think about using action filters in the future, just sayin'

Reply:
Done!